### PR TITLE
[CI] Run sample job YAML tests in buildkite

### PR DESCRIPTION
**Pull request by architkulkarni**
## Why are these changes needed?
Sets up the Buildkite CI pipeline to test the RayJob sample YAML files using `kind`.

## Related issue number

Closes #1246 
## Checks

- [ ] I've made sure the tests are passing. 
- Testing Strategy
   - [ ] Unit tests
   - [ ] Manual tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: tests/framework/config/kind-config-buildkite.yml [hunk 1]
```diff
@@ -14,4 +14,30 @@ nodes:
...
-        - "docker"
\ No newline at end of file
+        - "docker"
+  - |
+    kind: InitConfiguration
+    nodeRegistration:
+      kubeletExtraArgs:
+        node-labels: "ingress-ready=true"
+  extraPortMappings:
+    - containerPort: 30379
```
Comment (kevin85421) [inline tests/framework/config/kind-config-buildkite.yml:24]: There is no use case to expose the GCS port, so we can remove 6379 at this moment.

File: tests/framework/utils.py [hunk 2]
```diff
@@ -34,7 +35,12 @@ class CONST:
     # Paths
     REPO_ROOT = Path(__file__).absolute().parent.parent.parent
     HELM_CHART_ROOT = REPO_ROOT.joinpath("helm-chart")
-    DEFAULT_KIND_CONFIG = REPO_ROOT.joinpath("tests/framework/config/kind-config.yaml")
+
+    # Decide the config based on the environment
+    if os.getenv("BUILDKITE_ENV", default="") == "true":
```
Comment (kevin85421) [inline tests/framework/utils.py:40]: Why do we need a new kind config?

Comment (architkulkarni) [reply tests/framework/utils.py:40]: The new part of the config is copied from https://github.com/ray-project/ray/pull/22035/files#diff-cbbac00da1880527c8b203aeb03a1d3c492610ee1fc97f868392589c5b831893.  I'm not sure about the technical details here, but it's necessary to get `kubectl` to work in buildkite.

I'm not sure if it has any ill effects for users who want to run the test locally, so to be safe I just made another config file.

File: tests/framework/config/kind-config-buildkite.yml (hunk 1 above)
Comment (architkulkarni) [reply tests/framework/config/kind-config-buildkite.yml]: Done

File: tests/framework/utils.py (hunk 2 above)
Comment (kevin85421) [reply tests/framework/utils.py:40]: Got it. We can figure this out in the future.

**Review by kevin85421 — APPROVED** [review]
LGTM

**Comment by architkulkarni** [comment]
Not working yet so I'll mark as draft

**Comment by architkulkarni** [comment]
The Buildkite sample rayjob YAML test passed and the test logs are as expected: https://buildkite.com/ray-project/ray-ecosystem-ci-kuberay-ci/builds/1249#0189e5e3-9193-4717-8edd-c8c1bdc7ea3a/237-2560

**Pull request merged**